sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java

```diff
@@ -37,7 +37,7 @@
                 request.getResourceType(),
                 request.getOperationType());
 
-        AuthorizationTokenType originalRequestTokenType = request.authorizationTokenType;
+        AuthorizationTokenType originalRequestTokenType = authorizationTokenProvider.getAuthorizationTokenType();
```

Comment:
Please add tests.

Reply:
do we expect the original request and the barrier request to have different auth token types?

why can't we rely on the request.authTokenType?

Reply:
@milismsft  why was this resolved without adding any tests?

Reply:
BarrierRequest tests covering these changes already exist, adding a AAD E2E test case is out of scope since it requires permissions to the subscription which we do not have in CI.

I'm currently looking at extending one of the existing unit tests to add an expectation related assert or similar.

Reply:
Is there any reason a mocked test doing a BarrierRequest validating AAD auth be added?

Reply:
Hmm... if it only was that simple. Anyway, I've managed to get a test that provides direct coverage to the implementation changes.